Menu/PageTreeMenuBuilder.php

```diff
@@ -72,6 +72,12 @@ private function appendNodes (ItemInterface $parent, array $nodes, array $option
     {
         foreach ($nodes as $node)
         {
+            // Don't render nodes that the User is not supposed/eligible to see
```

Comment:
I think we don't have to touch the menu builder itself. The page needs to be in the page structure.

What we need to edit is the twig template, that does the rendering. Just skip the nodes there.